treatment-service/services/metric_service.go

```diff
@@ -7,34 +7,43 @@ import (
 	"strings"
 	"time"
 
-	"github.com/gojek/mlp/api/pkg/instrumentation/metrics"
-
 	"github.com/caraml-dev/xp/common/api/schema"
 	_segmenters "github.com/caraml-dev/xp/common/segmenters"
 	"github.com/caraml-dev/xp/treatment-service/config"
 	"github.com/caraml-dev/xp/treatment-service/instrumentation"
 	"github.com/caraml-dev/xp/treatment-service/models"
+	"github.com/gojek/mlp/api/pkg/instrumentation/metrics"
 )
 
 type MetricService interface {
+	LogFetchTreatmentMetrics(
+		begin time.Time,
+		projectId models.ProjectId,
+		treatment schema.SelectedTreatment,
+		requestFilter map[string][]*_segmenters.SegmenterValue,
+		statusCode int,
+	)
 	LogLatencyHistogram(begin time.Time, labels map[string]string, loggingMetric metrics.MetricName)
 	LogRequestCount(labels map[string]string, loggingMetric metrics.MetricName)
 
 	// GetProjectNameLabel retrieves only project name as labels
 	GetProjectNameLabel(projectId models.ProjectId) map[string]string
+	GetMetricLabels() []string
 	// GetLabels retrieves labels with flag to filter for segmenters
 	GetLabels(projectId models.ProjectId, treatment schema.SelectedTreatment, statusCode int,
-		metricLabels []string, requestFilter map[string][]*_segmenters.SegmenterValue, withSegmenters bool) map[string]string
+		requestFilter map[string][]*_segmenters.SegmenterValue, withSegmenters bool) map[string]string
+	ReplacePrometheusCollector(collector metrics.Collector)
 }
 
 type metricService struct {
 	Kind         config.MetricSinkKind
 	LocalStorage *models.LocalStorage
+	MetricLabels []string
 }
 
 func NewMetricService(cfg config.Monitoring, localStorage *models.LocalStorage) (MetricService, error) {
 	switch cfg.Kind {
-	case config.NoopMetricSink:
+	case config.NoopMetricSink, config.RPCMetricSink:
```

Comment:
The implementation for the RPC sink is still WIP, correct?

Reply:
Actually no, the `RPCMetricSink` is just a metric service type that informs the metric service to not initialise a new prometheus metrics collector, but to still use the logging methods `LogRequestCount` and `LogLatencyHistogram` as normal.

Reply:
Ok I think I got the difference between my assumption and the implementation now, after referring to the Turing PR: https://github.com/caraml-dev/turing/pull/302

Instead of the `RegisterMetricsCollector` call from Turing which effectively "sets" the collector on XP, can we have XP "get" the collector from Turing under the case `RPCMetricSink` here ? This would be a more natural flow, IMO. Wdyt?

Reply:
Hmm what you suggested is indeed more intuitive but I left the current implementation this way because of the order in which the metrics collector is received with respect to the creation of the experiment runner.

Currently, the app context (which includes the metric service) gets created during the creation of a new experiment runner (this happens in `NewExperimentRunner`). After this call returns, the newly created experiment runner then gets to 'set' the collector when the Turing Router calls `RegisterMetricsCollector`. 

In order to have the metric service set the collector when the metric service is being initialised (as part of the app context initialisation), we would need to somehow pass the metrics collector to whichever process is calling `NewExperimentRunner`, even before the experiment runner has been created. I'm not 100% sure right now how this would work but I can look into it.